tests/php/Control/UserDefinedFormControllerTest.php

```diff
@@ -467,4 +468,43 @@ public function testRecipientAttachment()
         $nodata = $this->findEmail('nodata@example.com', 'no-reply@example.com', 'Email Subject');
         $this->assertEmpty($nodata['AttachedFiles'], 'Recipients with HideFormData do not receive attachment');
     }
+
+    public function testMissingFolderCreated()
+    {
+        Config::modify()->set(Upload_Validator::class, 'use_is_uploaded_file', false);
+        $userForm = $this->setupFormFrontend('upload-form-without-folder');
+        $controller = UserDefinedFormController::create($userForm);
+        $field = $this->objFromFixture(EditableFileField::class, 'file-field-3');
+
+        $path = realpath(__DIR__ . '/fixtures/testfile.jpg');
+        $data = [
+            $field->Name => [
+                'name' => 'testfile.jpg',
+                'type' => 'image/jpeg',
+                'tmp_name' => $path,
+                'error' => 0,
+                'size' => filesize($path ?? ''),
+            ]
+        ];
+        $_FILES[$field->Name] = $data[$field->Name];
+
+        $controller->getRequest()->setSession(new Session([]));
+
+        $folderExistBefore = $field->getFolderExists();
+        $stage = Versioned::get_stage();
```

Comment:
```suggestion
        $stageBefore = Versioned::get_stage();
```
Makes it more obvious that you're testing that the stage was reset to what it was before